File: 28-06-2025/p2099.cs
public class Solution {
    public int[] MaxSubsequence(int[] nums, int k) {
        var h = new PriorityQueue<int, int>();

        // T: O(nlogk), S:O(k)
        for (var i = 0; i < nums.Length; ++i)
        {
            if (h.Count == k
                && h.TryDequeue(out var oldI, out var oldVal)
                && nums[i] < oldVal
            )
            {
                // O: T(logk)
                h.Enqueue(oldI, oldVal);
                continue;
            }

            // O: T(logk)
            h.Enqueue(i, nums[i]);
        }

        var h2 = new PriorityQueue<int, int>();

        // T: O(k), S:O(n)
        while (h.TryDequeue(out var i, out var _))
        {
            h2.Enqueue(i, i);
        }

        var result = new int[k];
        // T: O(k)
        for (var i = 0; i < k; ++i)
        {
            result[i] = nums[h2.Dequeue()];
        }

        return result;
    }
}

File: 28-06-2025/p2099.swift
class Solution {
  func maxSubsequence(_ nums: [Int], _ k: Int) -> [Int] {
    var storage: [[Int]] = []
    for i in 0..<nums.count {
      storage.append([i, nums[i]])
    }
    storage.sort(by: { $0[1] > $1[1] })
    var storage2: [[Int]] = []
    for i in 0..<k {
      storage2.append(storage[i])
    }
    storage2.sort(by: { $0[0] < $1[0] })
    var result: [Int] = []
    for i in 0..<storage2.count {
      result.append(storage2[i][1])
    }
    return result
  }
}